resolver/type_syntax.cc

```diff
@@ -312,6 +316,21 @@ ParsedSig parseSigWithSelfTypeParams(core::Context ctx, const ast::Send &sigSend
                     // was handled above
                     break;
                 case core::Names::abstract().rawId():
+                    if (auto e = ctx.beginError(sigSend.loc, core::errors::Resolver::InvalidMethodSignature)) {
+                        e.setHeader("Use `{}`, not `{}` to declare an abstract method", "sig(:abstract) { [...] }",
+                                    "sig { abstract.[...] }");
+
+                        auto loc = core::Loc(ctx.file, sigSend.loc);
+                        auto original = loc.source(ctx).value();
+                        auto corrected = original.replace(0, 3, "sig(:abstract)");
+                        auto abstractBegin = original.find("abstract.");
+                        corrected = corrected.erase(abstractBegin, 9);
+                        abstractBegin = original.find(".abstract");
+                        corrected = corrected.erase(abstractBegin, 9);
+
+                        e.replaceWith(fmt::format("Change `{}` to `{}`", original, corrected), loc, "{}", corrected);
```

Comment:
we should not make this an error in the same PR

we should first have a version that allows both syntaxes, and then have a version that has the autocorrect (two separate PRs + published rubygems versions)